src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -190,10 +200,34 @@ class EthService(
   }
 
   def submitHashRate(req: SubmitHashRateRequest): ServiceResponse[SubmitHashRateResponse] = {
-    //todo do we care about hash rate for now?
+    hashRate.updateAndGet(new UnaryOperator[Map[ByteString, (BigInt, Date)]] {
+      override def apply(t: Map[ByteString, (BigInt, Date)]): Map[ByteString, (BigInt, Date)] = {
+        val now = new Date
+        clearTimeouts(now, t + (req.id -> (req.hashRate, now)))
```

Comment:
Wait, hashrate is a number, so it should parsed as such. For example geth is fine with input like `123` (JSON int). This is what `extractQuantity` method should be used for.

Regarding `0x`, see comments on #195